packages/replay/src/util/addEvent.ts

```diff
@@ -39,5 +41,10 @@ export async function addEvent(
     replay.getContext().earliestEvent = timestampInMs;
   }
 
-  return replay.eventBuffer.addEvent(event, isCheckout);
+  try {
```

Comment:
This is the "actual" change. Note that for `finish` we already try-catch this (when flushing).